src/DeviceInput/InputDevices/internalDeviceSourceManager.ts

```diff
@@ -0,0 +1,179 @@
+import { Engine } from '../../Engines/engine';
+import { IDisposable } from '../../scene';
+import { DeviceType } from './deviceEnums';
+import { Nullable } from '../../types';
+import { Observable, Observer } from '../../Misc/observable';
+import { IDeviceEvent, IDeviceInputSystem } from './inputInterfaces';
+import { NativeDeviceInputSystem } from './nativeDeviceInputSystem';
+import { WebDeviceInputSystem } from './webDeviceInputSystem';
+import { DeviceSource } from './deviceSource';
+import { INative } from '../../Engines/Native/nativeInterfaces';
+
+declare const _native: INative;
+
+/** @hidden */

```

Comment:
I believe once you put `@hidden` the top level class, you don't need to add `@hidden` for each child public member, but this won't matter if we can do what I said in the other comment since this class will then be private.

Reply:
Talked offline